File: app/src/main/java/com/akexorcist/localizationapp/viewpager/ViewPagerActivity.java
package com.akexorcist.localizationapp.viewpager;

import android.os.Bundle;
import android.support.v4.view.ViewPager;
import android.view.View;
import android.widget.Button;

import com.akexorcist.localizationactivity.ui.LocalizationActivity;
import com.akexorcist.localizationapp.R;

public class ViewPagerActivity extends LocalizationActivity implements ViewPager.OnPageChangeListener {
    private final String KEY_CURRENT_PAGE = "current_page";

    private ViewPagerAdapter adapter;
    private ViewPager vpGreeting;
    private Button btnNext;
    private Button btnPrev;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_view_pager);

        findViewById(R.id.btn_america).setOnClickListener(onAmericaLanguageSelected());
        findViewById(R.id.btn_china).setOnClickListener(onChinaLanguageSelected());
        findViewById(R.id.btn_italy).setOnClickListener(onItalyLanguageSelected());
        findViewById(R.id.btn_japan).setOnClickListener(onJapanLanguageSelected());
        findViewById(R.id.btn_korea).setOnClickListener(onKoreaLanguageSelected());
        findViewById(R.id.btn_portugal).setOnClickListener(onPortugalLanguageSelected());
        findViewById(R.id.btn_thai).setOnClickListener(onThaiLanguageSelected());

        btnNext = findViewById(R.id.btn_next);
        btnNext.setOnClickListener(onNextPageClick());

        btnPrev = findViewById(R.id.btn_prev);
        btnPrev.setOnClickListener(onPreviousPageClick());

        adapter = new ViewPagerAdapter(getSupportFragmentManager());
        vpGreeting = findViewById(R.id.vp_greeting);
        vpGreeting.setAdapter(adapter);
        vpGreeting.addOnPageChangeListener(this);

        setNavigationVisibility(vpGreeting.getCurrentItem());
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        // Save current item of view pager.
        outState.putInt(KEY_CURRENT_PAGE, vpGreeting.getCurrentItem());
        super.onSaveInstanceState(outState);
    }

    @Override
    public void onRestoreInstanceState(Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);
        // Restore current item of view pager.
        vpGreeting.setCurrentItem(savedInstanceState.getInt(KEY_CURRENT_PAGE));
    }

    @Override
    public void onPageSelected(int position) {
        setNavigationVisibility(position);
    }

    @Override
    public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
    }

    @Override
    public void onPageScrollStateChanged(int state) {
    }

    private View.OnClickListener onAmericaLanguageSelected() {
        return view -> setLanguage("en");
    }

    private View.OnClickListener onChinaLanguageSelected() {
        return view -> setLanguage("zh");
    }

    private View.OnClickListener onItalyLanguageSelected() {
        return view -> setLanguage("it");
    }

    private View.OnClickListener onJapanLanguageSelected() {
        return view -> setLanguage("ja");
    }

    private View.OnClickListener onKoreaLanguageSelected() {
        return view -> setLanguage("ko");
    }

    private View.OnClickListener onPortugalLanguageSelected() {
        return view -> setLanguage("pt");
    }

    private View.OnClickListener onThaiLanguageSelected() {
        return view -> setLanguage("th");
    }

    private View.OnClickListener onNextPageClick() {
        return view -> vpGreeting.setCurrentItem(vpGreeting.getCurrentItem() + 1);
    }

    private View.OnClickListener onPreviousPageClick() {
        return view -> vpGreeting.setCurrentItem(vpGreeting.getCurrentItem() - 1);
    }

    private void setNavigationVisibility(int currentPosition) {
        btnPrev.setVisibility((currentPosition == 0) ? View.GONE : View.VISIBLE);
        btnNext.setVisibility((currentPosition == adapter.getCount() - 1) ? View.GONE : View.VISIBLE);
    }
}

File: app/src/main/java/com/akexorcist/localizationapp/viewpager/ViewPagerActivity.kt
package com.akexorcist.localizationapp.viewpager

import android.os.Bundle
import android.view.View
import androidx.viewpager.widget.ViewPager.OnPageChangeListener
import com.akexorcist.localizationactivity.ui.LocalizationActivity
import com.akexorcist.localizationapp.databinding.ActivityViewPagerBinding

class ViewPagerActivity : LocalizationActivity(), OnPageChangeListener {
    private val binding: ActivityViewPagerBinding by lazy { ActivityViewPagerBinding.inflate(layoutInflater) }
    private lateinit var adapter: ViewPagerAdapter

    companion object {
        private const val KEY_CURRENT_PAGE = "current_page"
    }

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)

        binding.btnBack.setOnClickListener { super.onBackPressed() }
        binding.layoutLanguageChooser.btnAmerican.setOnClickListener { setLanguage("en") }
        binding.layoutLanguageChooser.btnChinese.setOnClickListener { setLanguage("zh") }
        binding.layoutLanguageChooser.btnItalian.setOnClickListener { setLanguage("it") }
        binding.layoutLanguageChooser.btnJapanese.setOnClickListener { setLanguage("ja") }
        binding.layoutLanguageChooser.btnKorean.setOnClickListener { setLanguage("ko") }
        binding.layoutLanguageChooser.btnPortuguese.setOnClickListener { setLanguage("pt") }
        binding.layoutLanguageChooser.btnThai.setOnClickListener { setLanguage("th") }

        binding.btnNext.setOnClickListener { onNextPageClick() }
        binding.btnPrev.setOnClickListener { onPreviousPageClick() }

        adapter = ViewPagerAdapter(supportFragmentManager)
        binding.vpGreeting.adapter = adapter
        binding.vpGreeting.addOnPageChangeListener(this)
        setNavigationVisibility(binding.vpGreeting.currentItem)
    }

    public override fun onSaveInstanceState(outState: Bundle) {
        // Save current item of view pager.
        outState.putInt(KEY_CURRENT_PAGE, binding.vpGreeting.currentItem)
        super.onSaveInstanceState(outState)
    }

    public override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
        // Restore current item of view pager.
        binding.vpGreeting.currentItem = savedInstanceState.getInt(KEY_CURRENT_PAGE)
    }

    override fun onPageSelected(position: Int) {
        setNavigationVisibility(position)
    }

    override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {}
    override fun onPageScrollStateChanged(state: Int) {}

    private fun onNextPageClick() {
        binding.vpGreeting.currentItem = binding.vpGreeting.currentItem + 1
    }

    private fun onPreviousPageClick() {
        binding.vpGreeting.currentItem = binding.vpGreeting.currentItem - 1
    }

    private fun setNavigationVisibility(currentPosition: Int) {
        binding.btnPrev.visibility = if (currentPosition == 0) View.GONE else View.VISIBLE
        binding.btnNext.visibility = if (currentPosition == adapter.count - 1) View.GONE else View.VISIBLE
    }
}
